src/tooling/grain_doc.md

```diff
@@ -0,0 +1,132 @@
+
+Here's a quick example of adding a doc block to your grain code:
+```gr
+/**
+ * An alias for normal syntactic array access, i.e. `array[n]`.
+ *
+ * Retrieves the element from the array at the specified index.
+ * A negative index is treated as an offset from the end of the array.
+ *
+ * @param index: The index to access
+ * @param array: The array to access
+ * @returns The element from the array
+ *
+ * @since v0.1.0
+ * @history v0.2.0: Argument order changed to data-last
+ */
+export let get = (index, array) => {
+  array[index]
+}
+```
+
+## Usage
+
+Before running `grain doc`, you'll want to add doc block comments to your code. Doc block comments are started with `/**` and ended with `*/`. They must exist on the line above a type declaration or exported value. No docs will be generated for non-exported values so you can keep module internals private.
+
+You can run `grain doc <file|dir> -o <file|dir>` to generate markdown documentation for your code. As of Grain v0.5, the `grain doc` command supports directory input to recursively generate documentation for all `.gr` files.
+
+### Without doc blocks
+
+You can even use `grain doc` without doc blocks in your code. It will still generate simple documentation for data types and exported values, such as displaying their names and type signatures.
+
+## Annotations
+
+### @param
+
+```gr
+/**
+ * @param name: description
+ */
+```
+The `@param` annotation indicates a parameter to the function that the doc block is documenting.
+
+These should be specified in order of the function parameters and the name specified before the colon (`:`) should match the parameter name exactly—while not currently enforced, it will be required for named parameters in the future. The content after the colon is the description of the parameter.
+
+When generating docs, the type of the parameter will be detected by the compiler and inserted into the parameters table.
+
+### @returns
+
+```gr
+/**
+ * @returns description
+ */
+```
+
+The `@returns` annotation indicates the return of the function that the doc block is documenting.
+
+When generating docs, the type of the return will b detected by the compiler and inserted into the output.
+
+### @example
+
+```gr
+/**
+ * @example code
+ */
+```
+
+The `@example` annotation displaying how the function will be used. Currently only single line examples are allowed.
+
+### @since
+
+```gr
+/**
+ * @since version
+ */
+```
+
+The `@since` annotation specifies the [Semantic Version](https://semver.org/) the documenting export was added in.
+
+### @history
+```gr
+/**
+ * @history version description
+ */
+```
+
+The `@history` annotation similar to `@since` specifies the [Semantic Version](https://semvar.org/) a change to the documenting feature was made, multiple history annotations can be used in a single docblock. 
+
```

Comment:
```suggestion

When generating docs, will add a history table to the `<details>` element created by the `@since` attribute if both were used; otherwise, a new `<details>` element titled  "History" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "next" instead of the version.
```